// @runwayml/sdk/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.fileFromPath = exports.toFile = exports.UnprocessableEntityError = exports.PermissionDeniedError = exports.InternalServerError = exports.AuthenticationError = exports.BadRequestError = exports.RateLimitError = exports.ConflictError = exports.NotFoundError = exports.APIUserAbortError = exports.APIConnectionTimeoutError = exports.APIConnectionError = exports.APIError = exports.RunwayMLError = exports.RunwayML = void 0;
const Errors = __importStar(require("./error.js"));
const Uploads = __importStar(require("./uploads.js"));
const Core = __importStar(require("./core.js"));
const API = __importStar(require("./resources/index.js"));
/**
 * API Client for interfacing with the RunwayML API.
 */
class RunwayML extends Core.APIClient {
    /**
     * API Client for interfacing with the RunwayML API.
     *
     * @param {string | undefined} [opts.apiKey=process.env['RUNWAYML_API_SECRET'] ?? undefined]
     * @param {string | undefined} [opts.runwayVersion=2024-09-13]
     * @param {string} [opts.baseURL=process.env['RUNWAYML_BASE_URL'] ?? https://api.dev.runwayml.com] - Override the default base URL for the API.
     * @param {number} [opts.timeout=1 minute] - The maximum amount of time (in milliseconds) the client will wait for a response before timing out.
     * @param {number} [opts.httpAgent] - An HTTP agent used to manage HTTP(s) connections.
     * @param {Core.Fetch} [opts.fetch] - Specify a custom `fetch` function implementation.
     * @param {number} [opts.maxRetries=2] - The maximum number of times the client will retry a request.
     * @param {Core.Headers} opts.defaultHeaders - Default headers to include with every request to the API.
     * @param {Core.DefaultQuery} opts.defaultQuery - Default query parameters to include with every request to the API.
     */
    constructor({ baseURL = Core.readEnv('RUNWAYML_BASE_URL'), apiKey = Core.readEnv('RUNWAYML_API_SECRET'), runwayVersion = '2024-09-13', ...opts } = {}) {
        if (apiKey === undefined) {
            throw new Errors.RunwayMLError("The RUNWAYML_API_SECRET environment variable is missing or empty; either provide it, or instantiate the RunwayML client with an apiKey option, like new RunwayML({ apiKey: 'My API Key' }).");
        }
        const options = {
            apiKey,
            runwayVersion,
            ...opts,
            baseURL: baseURL || `https://api.dev.runwayml.com`,
        };
        if (Core.isRunningInBrowser()) {
            throw new Errors.RunwayMLError("It looks like you're running in a browser-like environment, which is disabled to protect your secret API credentials from attackers. If you have a strong business need for client-side use of this API, please open a GitHub issue with your use-case and security mitigations.");
        }
        super({
            baseURL: options.baseURL,
            timeout: options.timeout ?? 60000 /* 1 minute */,
            httpAgent: options.httpAgent,
            maxRetries: options.maxRetries,
            fetch: options.fetch,
        });
        this.tasks = new API.Tasks(this);
        this.imageToVideo = new API.ImageToVideo(this);
        this._options = options;
        this.apiKey = apiKey;
        this.runwayVersion = runwayVersion;
    }
    defaultQuery() {
        return this._options.defaultQuery;
    }
    defaultHeaders(opts) {
        return {
            ...super.defaultHeaders(opts),
            'X-Runway-Version': this.runwayVersion,
            ...this._options.defaultHeaders,
        };
    }
    authHeaders(opts) {
        return { Authorization: `Bearer ${this.apiKey}` };
    }
}
exports.RunwayML = RunwayML;
_a = RunwayML;
RunwayML.RunwayML = _a;
RunwayML.DEFAULT_TIMEOUT = 60000; // 1 minute
RunwayML.RunwayMLError = Errors.RunwayMLError;
RunwayML.APIError = Errors.APIError;
RunwayML.APIConnectionError = Errors.APIConnectionError;
RunwayML.APIConnectionTimeoutError = Errors.APIConnectionTimeoutError;
RunwayML.APIUserAbortError = Errors.APIUserAbortError;
RunwayML.NotFoundError = Errors.NotFoundError;
RunwayML.ConflictError = Errors.ConflictError;
RunwayML.RateLimitError = Errors.RateLimitError;
RunwayML.BadRequestError = Errors.BadRequestError;
RunwayML.AuthenticationError = Errors.AuthenticationError;
RunwayML.InternalServerError = Errors.InternalServerError;
RunwayML.PermissionDeniedError = Errors.PermissionDeniedError;
RunwayML.UnprocessableEntityError = Errors.UnprocessableEntityError;
RunwayML.toFile = Uploads.toFile;
RunwayML.fileFromPath = Uploads.fileFromPath;
exports.RunwayMLError = Errors.RunwayMLError, exports.APIError = Errors.APIError, exports.APIConnectionError = Errors.APIConnectionError, exports.APIConnectionTimeoutError = Errors.APIConnectionTimeoutError, exports.APIUserAbortError = Errors.APIUserAbortError, exports.NotFoundError = Errors.NotFoundError, exports.ConflictError = Errors.ConflictError, exports.RateLimitError = Errors.RateLimitError, exports.BadRequestError = Errors.BadRequestError, exports.AuthenticationError = Errors.AuthenticationError, exports.InternalServerError = Errors.InternalServerError, exports.PermissionDeniedError = Errors.PermissionDeniedError, exports.UnprocessableEntityError = Errors.UnprocessableEntityError;
exports.toFile = Uploads.toFile;
exports.fileFromPath = Uploads.fileFromPath;
(function (RunwayML) {
    RunwayML.Tasks = API.Tasks;
    RunwayML.ImageToVideo = API.ImageToVideo;
})(RunwayML = exports.RunwayML || (exports.RunwayML = {}));
exports = module.exports = RunwayML;
exports.default = RunwayML;
